app/models/compute_resources/foreman/model/gce.rb

```diff
@@ -154,7 +154,19 @@ def create_volumes(args)
     end
 
     def available_images
-      client.images
+      images_list = client.images.current
+      if image_families_to_filter.any?
+        regexp = Setting.convert_array_to_regexp(image_families_to_filter, prefix: '', suffix: '')
+        images_list.select! {|img| img.family&.match(regexp)}
+      end
+      images_list
+    end
+
+    # Includes the ability to override the images list from Google Compute Engine
+    # override this method to filter list by adding image family names
+    # Example - ['rhel', 'centos']
+    def image_families_to_filter
+      []
```

Comment:
Would it make sense to have some registration to this? Take it as nitpick, if this is not meant for major usage, no need for it now.
I.E like:
```ruby
  def self.image_families_to_filter
      @image_families_to_filter ||= []
  end
  def self.register_family_for_image_filter(family)
    image_families_to_filter << family
  end
```
```suggestion
      self.class.image_families_to_filter
```

Reply:
Done!